umbral/keys.py

```diff
@@ -58,82 +169,60 @@ def gen_key(cls, params: Optional[UmbralParameters] = None) -> 'UmbralPrivateKey
-        files. Unless you changed this when you called `to_bytes`, you should
-        not change it here. It is NOT recommended to change the `_scrypt_cost`
-        value unless you know what you're doing.
+        Optionally, uses a wrapping key to unwrap an encrypted Umbral private key.
+        Alternatively, if a password is provided it will derive the wrapping key
+        from it.
         """
         if params is None:
             params = default_params()
 
         if decoder:
             key_bytes = decoder(key_bytes)
 
-        if password:
-            salt = key_bytes[-16:]
-            key_bytes = key_bytes[:-16]
-
-            key = Scrypt(
-                salt=salt,
-                length=SecretBox.KEY_SIZE,
-                n=2**_scrypt_cost,
-                r=8,
-                p=1,
-                backend=default_backend()
-            ).derive(password)
-
-            key_bytes = SecretBox(key).decrypt(key_bytes)
+        if any((wrapping_key, password)):
+            key_bytes = unwrap_key(wrapped_key=key_bytes,
+                                   wrapping_key=wrapping_key,
+                                   password=password,
+                                   **kwargs)
 
         bn_key = CurveBN.from_bytes(key_bytes, params.curve)
         return cls(bn_key, params)
 
     def to_bytes(self,
+                 wrapping_key: Optional[bytes] = None,
                  password: Optional[bytes] = None,
-                 _scrypt_cost: int = 20,
-                 encoder: Optional[Callable] = None) -> bytes:
+                 encoder: Optional[Callable] = None,
+                 **kwargs) -> bytes:
         """
-        Returns an Umbral private key as bytes optional symmetric encryption
-        via nacl's Salsa20-Poly1305 and Scrypt key derivation. If a password
-        is provided, the user must encode it to bytes.
+        Returns an UmbralPrivateKey as bytes with optional symmetric
+        encryption via nacl's Salsa20-Poly1305.
+        If a password is provided instead of a wrapping key, it will use
+        Scrypt for key derivation.
         Optionally, allows an encoder to be passed in as a param to encode the
         data before returning it.
-
-        WARNING: RFC7914 recommends that you use a 2^20 cost value for sensitive
-        files. It is NOT recommended to change the `_scrypt_cost` value unless
-        you know what you are doing.
         """
-        umbral_privkey = self.bn_key.to_bytes()
 
-        if password:
-            salt = os.urandom(16)
-
-            key = Scrypt(
-                salt=salt,
-                length=SecretBox.KEY_SIZE,
-                n=2**_scrypt_cost,
-                r=8,
-                p=1,
-                backend=default_backend()
-            ).derive(password)
+        key_bytes = self.bn_key.to_bytes()
 
-            umbral_privkey = SecretBox(key).encrypt(umbral_privkey)
-            umbral_privkey += salt
+        if any((wrapping_key, password)):
```

Comment:
Why use `any` here?

Reply:
Because I want to call `wrap_key` only when `wrapping_key is not None or password is not None`. I guess `any((wrapping_key, password))` is shorter, but perhaps less clear?

Reply:
I was thinking 

```python
if wrapping_key or password:
```

Reply:
... yep.

Reply:
✅